alternator/streams.cc

```diff
@@ -993,9 +1001,13 @@ future<executor::request_return_type> executor::get_records(client_state& client
                 rjson::set(record, "eventName", "REMOVE");
                 break;
             }
-        }
-        if (limit > 0 && timestamp) {
-            maybe_add_record();
+            if (eor) {
+                maybe_add_record();
```

Comment:
I don't understand the change of this logic.
Previously, `maybe_add_record` was called every time `limit > 0` because `timestamp` was always set in line 943.
Now, `maybe_add_record` is only called when `eor` is true.

Could you please explain why this is ok @elcallio ? I'm probably missing something because I don't understand how `timestamp` could have ever been empty at this point.

Reply:
The diff is tricking you. There was two `maybe_add_record` calls before. One above, when checking if we switched timestamp, one post-loop that checked if we should seal the record being built (hoping for the best - that it was complete). 
Now there is only one, which after processing data in the cdc row, closes and adds the record iff `eor==true`, all inside the loop. We no longer have to even bother about the timestamp changing.